### PR TITLE
chore(submodule): update rule schema

### PR DESCRIPTION
**Pull request by brendongo**
PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [x] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure about any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

**Review by aryx — APPROVED** [review]

File: changelog.d/schema.changed [hunk 1]
```diff
@@ -0,0 +1 @@
+Rule validation no longer fails if a rule contains additional unknown fields. This makes it so older versions of semgrep do not fail rules that contain extra functionality. min-version field should be used to identify rules that should not be run (i.e. the additional functionality is necessary in running the rule)
```
Comment (mjambon) [inline changelog.d/schema.changed]: Are we not rejecting unknown fields in OCaml?

Edit: based on the test, I guess we're not.

Comment (mjambon) [reply changelog.d/schema.changed]: Please rename the changelog entry `gh-8712.changed` so it links to this PR.

File: cli/tests/e2e/test_rule_validation.py [hunk 2]
```diff
@@ -33,6 +33,26 @@ def test_validation_of_invalid_rules(run_semgrep_in_tmp: RunSemgrep, snapshot, r
...
+        output_format=OutputFormat.TEXT,
+        assert_exit_code={0},
+    )
+
+    snapshot.assert_match(
+        err,
+        "results.txt",
+    )
+
+
```
Comment (mjambon) [inline cli/tests/e2e/test_rule_validation.py:58]: A comment explaining what this tests would be welcome.

Comment (brendongo) [reply cli/tests/e2e/test_rule_validation.py:58]: done

**Review by mjambon — APPROVED** [review]

**Comment by mjambon** [comment]
lol... if only we had a check for unknown fields

image

I think in this case we're getting a new kind error. Not a problem. I'm regenerating the test output snapshots.

A spellchecker could be useful, though.

**Pull request merged**